core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -72,7 +72,7 @@ public static boolean isAncestorOf(Table table, long ancestorSnapshotId) {
   public static boolean isParentAncestorOf(
       Table table, long snapshotId, long ancestorParentSnapshotId) {
     for (Snapshot snapshot : ancestorsOf(snapshotId, table::snapshot)) {
-      if (snapshot.parentId() == ancestorParentSnapshotId) {
+      if (snapshot.parentId() != null && snapshot.parentId() == ancestorParentSnapshotId) {
```

Comment:
Can we add a test case for this scenario as well?

Reply:
SnapshotUtil doesn't have a test case class for it. So I created one, adding some additional test cases.